### PR TITLE
Update xpu related device setting

### PR DESCRIPTION
**Pull request by zhuhong61**
# What does this PR do?

This PR update some xpu related logic for correct support.

Fixes # (issue)

## Feature/Issue validation/testing

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
Logs for Test A

- [ ] Test B
Logs for Test B

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/llama-recipes/blob/main/CONTRIBUTING.md#pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes?  
- [ ] Did you write any new necessary tests?

Thanks for contributing 🎉!

### CODE REVIEW

**Comment by zhuhong61** [comment]
@abhilash1910 Could you please help review? Thanks!

**Review by abhilash1910 — APPROVED** [review]
Thanks for this update !
@HamidShojanazeri could you help take a look ?
cc @gujinghui

File: src/llama_recipes/utils/train_utils.py [hunk 1]
```diff
@@ -55,13 +58,15 @@ def train(model, train_dataloader,eval_dataloader, tokenizer, optimizer, lr_sche
     if train_config.use_fp16 and train_config.enable_fsdp:
         scaler = ShardedGradScaler()
     elif train_config.use_fp16 and not train_config.enable_fsdp:
-        scaler = torch.cuda.amp.GradScaler()
+        scaler = torch.xpu.amp.GradScaler() if is_xpu_available() else torch.cuda.amp.GradScaler()
```
Comment (gujinghui) [inline src/llama_recipes/utils/train_utils.py:61]: Do we really have torch.xpu.amp.GradScaler() already?

Comment (abhilash1910) [reply src/llama_recipes/utils/train_utils.py:61]: Yes I also had a doubt regarding this, not sure we have xpu.amp.GradScaler() .

Comment (zhuhong61) [reply src/llama_recipes/utils/train_utils.py:61]: > Do we really have torch.xpu.amp.GradScaler() already?

Thanks @gujinghui , @abhilash1910 , yes, we don't support torch.xpu.amp.GradScaler(), I will remove it and update it once we support this API.

Comment (gujinghui) [reply src/llama_recipes/utils/train_utils.py]: Should be yes? Assert to stop the workload with graceful exit message?

**Comment by BrodysgotMs** [comment]
$BYE

File: src/llama_recipes/utils/train_utils.py (hunk 1 above)
Comment (zhuhong61) [reply src/llama_recipes/utils/train_utils.py]: > Should be yes? Assert to stop the workload with graceful exit message?

Thanks for the suggestion. We have refined the code again, and could you please help review again? Thanks!

**Pull request closed without merging**